python/tests/unit/arcticdb/version_store/test_missing_empty.py

```diff
@@ -125,6 +125,39 @@ def pd_delete_replace(df1, df2, date_range=None):
     # concat preserves types over other methods eg int -> float (due to temp NaN creation)
     return pd_restore_secondary_index_levels(pd.concat(to_concat), df1.index.names)
 
+
+def infer_type(s: pd.Series):
+    t = None
+    for v in s:
+        if not pd.isnull(v):
+            tv = type(v)
+            if t and tv != t:
+                raise ValueError(f"Mixed non-None types in column {s.name}: {t} and {tv}")
+            t = tv
+    return t
+
+
+def fill_value(t):
+    fill_values = {
+        bool: False,
+        int: 0
+    }
+    if t in fill_values.keys():
+        return fill_values[t]
+    return None
+
+
+def pd_arcticdb_read_sim(df):
+    read_df = df.copy(True)
+    for c in df.columns:
+        t = infer_type(df[c])
+        print(c, t, fill_value(t))
+        if t and fill_value(t) is not None:
+            fv = fill_value(t)
+            read_df[c] = df[c].fillna(fv)
```

Comment:
nit: here it is enough to call `fill_value`  only once as follows
```python
fv = fill_value(t)
if fv: # if t is None fv will also be None 
    read_df[c] = df[c].fillna(fv)
```
In this case `fill_value` is not expensive but still, I would save the value in a variable and use it rather than re-calling the function.